sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -139,7 +139,9 @@ case class AdaptiveSparkPlanExec(
   }
 
   private def getFinalPhysicalPlan(): SparkPlan = lock.synchronized {
-    if (!isFinalPlan) {
+    if (isFinalPlan) return currentPhysicalPlan
+
+    context.session.withActive {
```

Comment:
Maybe add a comment why this is needed. Something in the spirit of `Adaptive execution can replan a query outside of withActive scope functions (e.g. plan.queryExecution.rdd and that is why we need to set it here.`